src/accessibility/browser.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  getRole,
+  computeAccessibleName,
+  computeAccessibleDescription,
+} from 'dom-accessibility-api';
+import type { AccessibilityTreeOptions } from '.';
+
+const indent = (text: string, indenter = '  ') =>
+  indenter + text.split('\n').join(`\n${indenter}`);
+
+const enum AccessibilityState {
+  /** This element is accessible and its descendents could be accessible */
+  Accessible,
+  /** This element is innaccessible, but its descendents could be accessible */
+  InaccessibleSelf,
+  /** Both this element and its descendents are inaccessible */
+  InaccessibleSelfAndDescendents,
+}
+
+// TODO in PR: what about role="presentation" or role="none"? Should they be excluded?
+// The spec says "it depends on the implementation" basically I think
+const getAccessibilityState = (element: Element): AccessibilityState => {
```

Comment:
Nit: I wonder if the function name could be changed to make it clearer that it gets the state of a single element? `getElementAccessibility` or something?